src/request_rate_manager.cc

```diff
@@ -283,7 +283,10 @@ RequestRateManager::ResumeWorkers()
   start_time_ = std::chrono::steady_clock::now();
 
   // Wake up all the threads to begin execution
-  execute_ = true;
+  {
+    std::lock_guard<std::mutex> lock(wake_mutex_);
+    execute_ = true;
+  }
```

Comment:
This just seemed like an uncaught potential race condition to me, when looking at how the [condition variable docs](https://en.cppreference.com/w/cpp/thread/condition_variable) say it should be used.

There are likely other race conditions still present but hopefully this removes one, and CI did pass.